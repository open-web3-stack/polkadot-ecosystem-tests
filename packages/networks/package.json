{
	"name": "@e2e-test/networks",
	"type": "module",
	"dependencies": {
		"@acala-network/chopsticks-testing": "^0.13.1",
		"@e2e-test/shared": "workspace:*"
	},
	"main": "./dist/cjs/index.js",
	"module": "./dist/esm/index.js",
	"files": [
		"*",
		"!tsconfig.json",
		"!**/*.test.ts"
	],
	"exports": {
		".": {
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./chains": {
			"require": "./dist/cjs/chains/index.js",
			"import": "./dist/esm/chains/index.js",
			"default": "./dist/esm/chains/index.js"
		},
		"./*": {
			"require": "./dist/cjs/*.js",
			"import": "./dist/esm/*.js",
			"default": "./dist/esm/*.js"
		}
	}
}
