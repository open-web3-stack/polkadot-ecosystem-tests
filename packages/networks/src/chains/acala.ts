import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  acala: {
    relayToken: 'DOT',
    relayLiquidToken: 'LDOT',
    stableToken: 'AUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    dot: { Token: 'DOT' },
    ldot: { Token: 'LDOT' },
    dai: { Erc20: '0x54a37a01cd75b616d63e0ab665bffdb0143c52ae' },
    wbtc: { ForeignAsset: 5 },
    ausd: { Token: 'AUSD' },
    aca: { Token: 'ACA' },
    lcdot: { LiquidCrowdloan: 13 },
  },
  karura: {
    relayToken: 'KSM',
    relayLiquidToken: 'LKSM',
    stableToken: 'KUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    ksm: { Token: 'KSM' },
    lksm: { Token: 'LKSM' },
    usdt: { ForeignAsset: 7 },
    rmrk: { ForeignAsset: 0 },
    dai: { Erc20: '0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71' },
    ausd: { Token: 'KUSD' },
    kar: { Token: 'KAR' },
  },
}

const getInitStorages = (config: typeof custom.acala | typeof custom.karura) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 4, data: { free: 10 * 1e12 } }]],
  },
  Tokens: {
    accounts: [
      [[defaultAccount.alice.address, { Token: config.relayToken }], { free: 10 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.relayLiquidToken }], { free: 100 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.stableToken }], { free: 1000 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  EvmAccounts: {
    accounts: [[['0x82a258cb20e2adb4788153cd5eb5839615ece9a0'], defaultAccount.alice.address]],
    evmAddresses: [[[defaultAccount.alice.address], '0x82a258cb20e2adb4788153cd5eb5839615ece9a0']],
  },
  Homa: {
    // avoid impact test outcome
    $removePrefix: ['redeemRequests', 'unbondings', 'toBondPool'],
    // so that bump era won't trigger unbond
    relayChainCurrentEra: 100,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const acala = defineChain({
  name: 'acala',
  endpoint: 'wss://acala-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.acala,
  initStorages: getInitStorages(custom.acala),
})

export const karura = defineChain({
  name: 'karura',
  endpoint: 'wss://karura-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.karura,
  initStorages: getInitStorages(custom.karura),
})
