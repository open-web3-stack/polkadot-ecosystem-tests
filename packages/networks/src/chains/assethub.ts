import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
  assetHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubPolkadot | typeof custom.assetHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubPolkadot = defineChain({
  name: 'assetHubPolkadot',
  endpoint: 'wss://rpc-asset-hub-polkadot.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubPolkadot,
  initStorages: getInitStorages(custom.assetHubPolkadot),
})

export const assetHubKusama = defineChain({
  name: 'assetHubKusama',
  endpoint: 'wss://rpc-asset-hub-kusama.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubKusama,
  initStorages: getInitStorages(custom.assetHubKusama),
})
