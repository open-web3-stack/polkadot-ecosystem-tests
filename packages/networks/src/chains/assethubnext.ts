import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNextWestend: {
    wnd: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNextWestend) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNextWestend = defineChain({
  name: 'assetHubNextWestend',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNextWestend,
  initStorages: getInitStorages(custom.assetHubNextWestend),
})
