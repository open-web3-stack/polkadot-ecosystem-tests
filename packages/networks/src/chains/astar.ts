import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  astar: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551617',

    paraAccount: '13YMK2eZzuFY1WZGagpYtTgbWBWGdoUD2CtrPj1mQPjY8Ldc',
    dot: 340282366920938463463374607431768211455n,
    astr: { Concrete: { parents: 0, interior: 'Here' } },
    aca: 18446744073709551616n,
    usdt: 4294969280n,
  },
  shiden: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551616',

    paraAccount: 'F7fq1jNy74AqkJ1DP4KqSrWtnTGtXfNVoDwFhTvvPxUvJaq',
    ksm: 340282366920938463463374607431768211455n,
    sdn: { Concrete: { parents: 0, interior: 'Here' } },
    kar: 18446744073709551618n,
    usdt: 4294969280n,
  },
}

const getInitStorages = (config: typeof custom.astar | typeof custom.shiden) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: '100000000000000000000' } }]],
  },
  Assets: {
    account: [
      [[config.relayToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
      [[config.aUSDToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccounts.alice.address,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const astar = defineChain({
  name: 'astar',
  paraId: 2006,
  endpoint: 'wss://astar-rpc.dwellir.com',
  custom: custom.astar,
  initStorages: getInitStorages(custom.astar),
})

export const shiden = defineChain({
  name: 'shiden',
  paraId: 2007,
  endpoint: 'wss://shiden-rpc.dwellir.com',
  custom: custom.shiden,
  initStorages: getInitStorages(custom.shiden),
})
