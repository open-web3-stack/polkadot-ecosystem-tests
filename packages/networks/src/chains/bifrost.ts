import { defineChain } from '../defineChain.js'
import { defaultAccounts, defaultAccountsSr25519, testAccounts } from '../testAccounts.js'

const custom = {
  bifrostPolkadot: {
    relayToken: 'DOT',
    bnc: { Native: 'BNC' },
  },
  bifrostKusama: {
    relayToken: 'KSM',
    bnc: { Token: 'BNC' },
  },
}

const getInitStorages = (_config: typeof custom.bifrostPolkadot | typeof custom.bifrostKusama) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[testAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
    ],
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers'],
  },
})

export const bifrostPolkadot = defineChain({
  name: 'bifrostPolkadot',
  endpoint: 'wss://bifrost-polkadot.ibp.network',
  paraId: 2030,
  custom: custom.bifrostPolkadot,
  initStorages: getInitStorages(custom.bifrostPolkadot),
})

export const bifrostKusama = defineChain({
  name: 'bifrostKusama',
  endpoint: 'wss://us.bifrost-rpc.liebi.com/ws',
  paraId: 2001,
  custom: custom.bifrostKusama,
  initStorages: getInitStorages(custom.bifrostKusama),
})
