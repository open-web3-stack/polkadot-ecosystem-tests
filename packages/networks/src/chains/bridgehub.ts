import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  bridgeHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
  },
  bridgeHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
  },
}

const getInitStorages = (_config: typeof custom.bridgeHubPolkadot | typeof custom.bridgeHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
})

export const bridgeHubPolkadot = defineChain({
  name: 'bridgeHubPolkadot',
  endpoint: 'wss://polkadot-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubPolkadot,
  initStorages: getInitStorages(custom.bridgeHubPolkadot),
})

export const bridgeHubKusama = defineChain({
  name: 'bridgeHubKusama',
  endpoint: 'wss://kusama-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubKusama,
  initStorages: getInitStorages(custom.bridgeHubKusama),
})
