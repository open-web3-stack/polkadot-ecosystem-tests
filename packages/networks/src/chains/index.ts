export * from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
