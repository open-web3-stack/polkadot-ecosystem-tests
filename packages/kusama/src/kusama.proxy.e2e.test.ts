import { kusama } from '@e2e-test/networks/chains'
import {
  createProxyConfig,
  defaultProxyTypeConfig,
  fullProxyE2ETests,
  KusamaProxyTypes,
  type ProxyTestConfig,
  type ProxyTypeConfig,
  type RelayTestConfig,
  registerTestTree,
} from '@e2e-test/shared'

const testConfig: RelayTestConfig = {
  testSuiteName: 'Kusama Proxy',
  addressEncoding: 2,
  blockProvider: 'Local',
}

/**
 * When removing a proxy account, the calling account's proxy type needs to be at least as powerful as the
 * proxy type of the account being removed.
 *
 * The lattice that relay chain proxy types form is such that
 * 1. `Any` is the maximum
 * 2. `NonTransfer` is, barring `Any`, the supremum (least upper bound)of all other types
 * 3. there is no infimum between the remaining types
 *
 * Therefore, to test proxy removal with a `ParaRegistration` proxy type, the proxy type to be removed
 * must be `ParaRegistration`.
 *
 * Thus, relaychains' proxy E2E tests need to provide a custom `ParaRegistration` proxy type config
 * that specifies the type being passed to the proxy removal action builder.
 */
const kusamaProxyTypeConfig: ProxyTypeConfig = {
  ...defaultProxyTypeConfig,
  ['Any']: {
    buildAllowedActions: (builder) => [
      ...builder.buildBalancesAction(),
      ...builder.buildMultisigAction(),
      ...builder.buildProxyAction(),
      ...builder.buildProxyRejectAnnouncementAction(),
      ...builder.buildProxyRemovalAction(KusamaProxyTypes.Any),
      ...builder.buildSystemAction(),
      ...builder.buildUtilityAction(),
    ],
    buildDisallowedActions: (builder) => [
      ...builder.buildAuctionAction(),
      ...builder.buildBountyAction(),
      ...builder.buildGovernanceAction(),
      ...builder.buildNominationPoolsAction(),
      ...builder.buildStakingAction(),
      ...builder.buildVestingAction(),
    ],
  },
  ['NonTransfer']: {
    buildAllowedActions: (builder) => [
      ...builder.buildProxyAction(),
      ...builder.buildMultisigAction(),
      ...builder.buildSystemRemarkAction(),
      ...builder.buildUtilityAction(),
    ],
    buildDisallowedActions: (builder) => [
      ...builder.buildAuctionAction(),
      ...builder.buildBalancesAction(),
      ...builder.buildBountyAction(),
      ...builder.buildGovernanceAction(),
      ...builder.buildNominationPoolsAction(),
      ...builder.buildStakingAction(),
      ...builder.buildVestingAction(),
    ],
  },

  ['Auction']: {
    buildAllowedActions: (builder) => [...builder.buildCrowdloanAction(), ...builder.buildParasRegistrarAction()],
    buildDisallowedActions: (builder) => [
      ...builder.buildBalancesAction(),
      ...builder.buildStakingAction(),
      ...builder.buildSystemAction(),
      ...builder.buildGovernanceAction(),
      ...builder.buildVestingAction(),

      // TODO: this is not right, see https://github.com/polkadot-fellows/runtimes/blob/main/relay/kusama/src/lib.rs#L1391
      // Needs an issue
      ...builder.buildAuctionAction(),
      ...builder.buildSlotsAction(),
    ],
  },

  ['Governance']: {
    buildAllowedActions: (builder) => [...builder.buildUtilityAction()],
    buildDisallowedActions: (builder) => [
      ...builder.buildBalancesAction(),
      ...builder.buildBountyAction(),
      ...builder.buildGovernanceAction(),
      ...builder.buildMultisigAction(),
      ...builder.buildProxyAction(),
      ...builder.buildStakingAction(),
      ...builder.buildSystemAction(),
      ...builder.buildVestingAction(),
    ],
  },

  ['Staking']: {
    buildAllowedActions: (builder) => [...builder.buildUtilityAction()],
    buildDisallowedActions: (builder) => [
      ...builder.buildBalancesAction(),
      ...builder.buildFastUnstakeAction(),
      ...builder.buildGovernanceAction(),
      ...builder.buildNominationPoolsAction(),
      ...builder.buildStakingAction(),
      ...builder.buildSystemAction(),
      ...builder.buildVestingAction(),
    ],
  },

  ['NominationPools']: {
    buildAllowedActions: (builder) => [...builder.buildUtilityAction()],
    buildDisallowedActions: (builder) => [
      ...builder.buildBalancesAction(),
      ...builder.buildGovernanceAction(),
      ...builder.buildNominationPoolsAction(),
      ...builder.buildStakingAction(),
      ...builder.buildSystemAction(),
      ...builder.buildVestingAction(),
    ],
  },
  ['Society']: {
    buildAllowedActions: (_builder) => [],
    buildDisallowedActions: (builder) => [
      ...builder.buildBalancesAction(),
      ...builder.buildSocietyAction(),
      ...builder.buildSystemAction(),
      ...builder.buildUtilityAction(),
    ],
  },
  ['ParaRegistration']: {
    buildAllowedActions: (builder) => [
      ...builder.buildParasRegistrarAction(),
      ...builder.buildUtilityAction(),
      ...builder.buildProxyRemovalAction(KusamaProxyTypes.ParaRegistration),
    ],
    buildDisallowedActions: (builder) => [...defaultProxyTypeConfig.ParaRegistration.buildDisallowedActions(builder)],
  },
}

const kusamaProxyCfg: ProxyTestConfig = createProxyConfig(KusamaProxyTypes, kusamaProxyTypeConfig)

registerTestTree(fullProxyE2ETests(kusama, testConfig, kusamaProxyCfg))
