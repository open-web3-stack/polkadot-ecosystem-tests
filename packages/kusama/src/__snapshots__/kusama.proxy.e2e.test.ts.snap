// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Kusama Proxy > add proxies (with/without delay) to an account, and remove them > events when adding proxies to Alice 1`] = `
[
  {
    "data": {
      "delay": 0,
      "delegatee": "CuW7GM1JCXTG5Whmw7yV53Wi4trhqRFmDC49NSc4ZQ36UCg",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Any",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "Ftk6uzZmZVciEaSMHBZ3b1272KavruU4NxLjGLQVsUAKV3B",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "NonTransfer",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "Ef4fPbsLDyyhBZMsLNxuErttrTVeZygduXM26bupXbrgjKP",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Governance",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "Fju5xS5KrXfrYCDMRPZa38t7x73jumfQsTqxpFgjni58nXF",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Staking",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "FF84pi37FUToJKQSPaMDYsY96BLNzhyLQ5opd2AWFu7HCKe",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "CancelProxy",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "F1yN2phuNDeehafhtzb9bJmkdShgEuMqxwZbRKMaeAqvUfU",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Auction",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "HuzD9NcQaDgcMcSYWpeAPFDaLC7dS53TDruPCmU97njVEzZ",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Society",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "E2SeX2FCET9jK6AgYaz2bjMsjp8DdE5BRchooMp5uMBoL9B",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "NominationPools",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "DgBeKgakkePrTGGSi4deeNAiZzPRJ8mszSZqmtp5CYWm9JY",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Spokesperson",
    },
    "method": "ProxyAdded",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > add proxies (with/without delay) to an account, and remove them > events when removing all proxies from Alice 1`] = `[]`;

exports[`Kusama Proxy > add proxies (with/without delay) to an account, and remove them > events when removing proxies from Alice (batch) 1`] = `
[
  {
    "data": {
      "delay": 0,
      "delegatee": "CuW7GM1JCXTG5Whmw7yV53Wi4trhqRFmDC49NSc4ZQ36UCg",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Any",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "Ftk6uzZmZVciEaSMHBZ3b1272KavruU4NxLjGLQVsUAKV3B",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "NonTransfer",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "Ef4fPbsLDyyhBZMsLNxuErttrTVeZygduXM26bupXbrgjKP",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Governance",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "Fju5xS5KrXfrYCDMRPZa38t7x73jumfQsTqxpFgjni58nXF",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Staking",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "FF84pi37FUToJKQSPaMDYsY96BLNzhyLQ5opd2AWFu7HCKe",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "CancelProxy",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "F1yN2phuNDeehafhtzb9bJmkdShgEuMqxwZbRKMaeAqvUfU",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Auction",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "HuzD9NcQaDgcMcSYWpeAPFDaLC7dS53TDruPCmU97njVEzZ",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Society",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "E2SeX2FCET9jK6AgYaz2bjMsjp8DdE5BRchooMp5uMBoL9B",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "NominationPools",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
  {
    "data": {
      "delay": 0,
      "delegatee": "DgBeKgakkePrTGGSi4deeNAiZzPRJ8mszSZqmtp5CYWm9JY",
      "delegator": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "proxyType": "Spokesperson",
    },
    "method": "ProxyRemoved",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > create and kill pure proxies > events when creating pure proxies for Alice 1`] = `
[
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "Any",
      "who": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "NonTransfer",
      "who": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "Governance",
      "who": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "Staking",
      "who": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "CancelProxy",
      "who": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "Auction",
      "who": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "Society",
      "who": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "NominationPools",
      "who": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
  {
    "data": {
      "disambiguationIndex": 0,
      "proxyType": "Spokesperson",
      "who": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
    },
    "method": "PureCreated",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > create and kill pure proxies > events when killing pure proxy of type 0 for Alice 1`] = `
[
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > create and kill pure proxies > events when killing pure proxy of type 1 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > create and kill pure proxies > events when killing pure proxy of type 2 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > create and kill pure proxies > events when killing pure proxy of type 3 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > create and kill pure proxies > events when killing pure proxy of type 5 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > create and kill pure proxies > events when killing pure proxy of type 6 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > create and kill pure proxies > events when killing pure proxy of type 7 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > create and kill pure proxies > events when killing pure proxy of type 8 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > create and kill pure proxies > events when killing pure proxy of type 9 for Alice 1`] = `
[
  {
    "data": {
      "result": {
        "Err": {
          "Module": {
            "error": "0x05000000",
            "index": 0,
          },
        },
      },
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;

exports[`Kusama Proxy > perform proxy call on behalf of delegator > events when Bob transfers funds to Charlie as Alice's proxy 1`] = `
[
  {
    "data": {
      "amount": 1000000000000,
      "from": "HNZata7iMYWmk5RvZRTiAsSDhV8366zq2YGb3tLH5Upf74F",
      "to": "Fr4NzY1udSFFLzb2R3qxVQkwz9cZraWkyfH4h3mVVk7BK7P",
    },
    "method": "Transfer",
    "section": "balances",
  },
  {
    "data": {
      "result": "Ok",
    },
    "method": "ProxyExecuted",
    "section": "proxy",
  },
]
`;
